examples/quad/main.rs

```diff
@@ -727,13 +509,195 @@ fn swapchain_stuff(
         pipeline.unwrap()
     };
 
-    (
-        swap_chain,
-        render_pass,
-        framebuffers,
-        frame_images,
-        pipeline,
-        pipeline_layout,
-        extent,
-    )
+    // Rendering setup
+    let mut viewport = pso::Viewport {
+        rect: pso::Rect {
+            x: 0,
+            y: 0,
+            w: extent.width as _,
+            h: extent.height as _,
+        },
+        depth: 0.0..1.0,
+    };
+
+    //
+    let mut running = true;
+    let mut recreate_swapchain = false;
+    while running {
+        events_loop.poll_events(|event| {
+            if let winit::Event::WindowEvent { event, .. } = event {
+                #[allow(unused_variables)]
+                match event {
+                    winit::WindowEvent::KeyboardInput {
+                        input:
+                            winit::KeyboardInput {
+                                virtual_keycode: Some(winit::VirtualKeyCode::Escape),
+                                ..
+                            },
+                        ..
+                    }
+                    | winit::WindowEvent::CloseRequested => running = false,
+                    winit::WindowEvent::Resized(dims) => {
+                        println!("resized to {:?}", dims);
+                        #[cfg(feature = "gl")]
+                        surface.get_window().resize(dims.to_physical(surface.get_window().get_hidpi_factor()));
+                        recreate_swapchain = true;
+                    }
+                    _ => (),
+                }
+            }
+        });
+
+        if recreate_swapchain {
+            device.wait_idle().unwrap();
+
+            let (caps, formats, _present_modes) = surface.compatibility(&mut adapter.physical_device);
```

Comment:
Since we aren't re-creating the pipeline state and render passes here, let's just `assert_eq!` on the old format being available.

Reply:
like this? 
```rust
  let (caps, formats, _present_modes) = surface.compatibility(&mut adapter.physical_device);
  assert_eq!(formats.iter().any(|fs| fs.iter().any(|f| *f == format)),true);
```

Reply:
yes, but just `assert!(x)` instead of `assert_eq!(x, true)` :)

Reply:
ha, of course.